### PR TITLE
FIX: Don't show that an existing user is invited_by another user

### PR DESCRIPTION
**Pull request by nattsw**
If an existing user (John) accepts an invite created by Kenny to a group, John may be seen as invited by Kenny, despite already having an account on the site.

This PR removes the bug by excluding invites that determine the `invited_by` after the user's creation date. The delay buffer in the query accounts for invites that also create the user at the same time.

### CODE REVIEW

**Comment by nattsw** [comment]
Test failure looks legit.

File: app/models/user.rb [hunk 1]
```diff
@@ -602,8 +602,17 @@ def sync_notification_channel_position
...
+    invite_redemption_delay = 5.seconds
     used_invite =
-      Invite.with_deleted.joins(:invited_users).where("invited_users.user_id = ?", self.id).first
+      Invite
+        .with_deleted
+        .joins(:invited_users)
+        .where(
+          "invited_users.user_id = ? AND invited_users.redeemed_at <= ?",
+          self.id,
+          self.created_at + invite_redemption_delay,
```
Comment (tgxworld) [inline app/models/user.rb:613]: I don't quite understand this. If the user was not invited by the user given by `invited_users.user_id`, why does the `invited_users` record exist? 🤔 Maybe the `invited_users` record association should not exist in the first place?

Comment (nattsw) [reply app/models/user.rb:613]: From my understanding, the `invited_users` record gets created to link the inviter to the invitee. 

Looking at history, it looks like `invited_users` is a method to allow tallying `invite.max_redemptions_allowed`.

Can you elaborate on what you don't understand?

Comment (tgxworld) [reply app/models/user.rb:613]: > If an existing user (John) accepts an invite created by Kenny to a group, John may be seen as invited by Kenny, despite already having an account on the site.

This is the first statement which I don't understand. If John exists, he should just be added to the group without being tied to Kenny as the inviter through a `invited_users` record. If feels like the problem is somewhere else when reading through what this PR is trying to do. 

My main worry here is the use of a delay buffer which we cannot accuratly determine what the actual value should be.

Comment (nattsw) [reply app/models/user.rb:613]: I agree with the problem of the delay buffer. It was to cater for cases where user accounts that were created with an invite. I'm not sure if there's a better way around this.

I think it still makes sense for an `invited_users` record though. If Kenny creates an invite for 30 redemptions, I feel like there should be 30 records of `invited_users` when they are all redeemed.

**Review by tgxworld — APPROVED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (nattsw) [reply app/models/user.rb:616]: TGX and I had a chat about this implementation. Due to the fact that users created via invites are created **after** the invite is redeemed, we have to deal with this delay.

**Pull request merged**